elastic_search/main.tf

```diff
@@ -145,73 +113,60 @@ resource "aws_cloudwatch_log_resource_policy" "example" {
 CONFIG
 }
 
-resource "aws_elasticsearch_domain" "variant-elk-cluster" {
-    domain_name           = var.domain_name
-    elasticsearch_version = var.es_version
-    domain_endpoint_options {
-        enforce_https=true
-        tls_security_policy = "Policy-Min-TLS-1-2-2019-07"
-    }
-    advanced_security_options {
-        enabled = true
-        internal_user_database_enabled = true
-        master_user_options {
-            master_user_name = var.master_user_options["master_user_name"]
-            master_user_password = lookup( var.master_user_options, "master_user_password", random_password.password.result)
-        }
-    }
-    ebs_options {
-      ebs_enabled = true
-      volume_size = var.ebs_volume_size
-    }
-    encrypt_at_rest {
-        enabled = true
-    }
-    node_to_node_encryption {
-        enabled = true
+resource "aws_elasticsearch_domain" "cluster" {
+  domain_name           = var.domain_name
+  elasticsearch_version = var.es_version
+  domain_endpoint_options {
+    enforce_https       = true
+    tls_security_policy = "Policy-Min-TLS-1-2-2019-07"
+  }
+  advanced_security_options {
+    enabled                        = true
+    internal_user_database_enabled = true
+    master_user_options {
+      master_user_name     = var.master_user_options["master_user_name"]
+      master_user_password = lookup(var.master_user_options, "master_user_password", random_password.password.result)
     }
-    dynamic "cluster_config" {
-        for_each = var.cluster_config["dedicated_master_enabled"] == true ? [1] : []
-        content {
-            instance_type = contains("instance_type" , var.cluster_config) ? var.cluster_config["instance_type"] : "r5.large.elasticsearch"
-            instance_count = local.instance_count
-            dedicated_master_enabled = var.cluster_config["dedicated_master_enabled"]
-            dedicated_master_count = var.cluster_config["dedicated_master_count"]
-            zone_awareness_enabled = true
-            zone_awareness_config {
-                availability_zone_count = 3
-            }
-        }
+  }
+  ebs_options {
+    ebs_enabled = true
+    volume_size = var.ebs_volume_size
+  }
+  encrypt_at_rest {
+    enabled = true
+  }
+  node_to_node_encryption {
+    enabled = true
+  }
 
+  cluster_config {
+    instance_type            = local.instance_type
+    instance_count           = local.instance_count
+    dedicated_master_enabled = var.cluster_config["dedicated_master_enabled"] ? var.cluster_config["dedicated_master_enabled"] : false
```

Comment:
Isn't this just a long way of writing `var.cluster_config["dedicated_master_enabled"]`? Or is this necessary because that value could be null? To me this just looks like:
`true ? true : false` or `false ? true: false`

Reply:
I think I missed to add == null. I'll fix this